src/Chapter1.hs

```diff
@@ -553,8 +551,11 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | ((x > y) && (y > z)) || ((x < y) && (y < z)) = y
+    | ((y > x) && (x > z)) || ((y < x) && (x < z)) = x
+    | ((y > x) && (z > x)) || ((y < x) && (z < x)) = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂
But as you noticed, these cases cover all possible cases, so if you didn't fit the first two patterns, you know for sure that the answer is `z`, you can even skip checking it:

```suggestion
    | otherwise = z
```